docs/vgclog.md

```diff
@@ -0,0 +1,423 @@
+
+  <ownableSynchronizers candidates="6711" cleared="2579" /> 
+
+  <references type="soft" candidates="9233" cleared="0" enqueued="0" dynamicThreshold="32" maxThreshold="32" /> 
+
+  <references type="weak" candidates="13149" cleared="4781" enqueued="4495" /> 
+
+  <references type="phantom" candidates="359" cleared="8" enqueued="8" /> 
+
+  <object-monitors candidates="79" cleared="36"  /> 
+
+</gc-op> 
+
+<gc-end id="8" type="scavenge" contextid="4" durationms="30.815" usertimems="89.130" systemtimems="3.179" stalltimems="1.944" timestamp="2020-10-18T13:27:09.634" activeThreads="4"> 
+
+  <mem-info id="9" free="910816000" total="1073741824" percent="84"> 
+
+    <mem type="nursery" free="108247360" total="268435456" percent="40"> 
+
+      <mem type="allocate" free="108247360" total="134217728" percent="80" /> 
+
+      <mem type="survivor" free="0" total="134217728" percent="0" /> 
+
+    </mem> 
+
+    <mem type="tenure" free="802568640" total="805306368" percent="99"> 
+
+      <mem type="soa" free="762302912" total="765040640" percent="99" /> 
+
+      <mem type="loa" free="40265728" total="40265728" percent="100" /> 
+
+    </mem> 
+
+    <pending-finalizers system="1" default="156" reference="4503" classloader="0" /> 
+
+    <remembered-set count="30758" /> 
+
+  </mem-info> 
+
+</gc-end> 
+
+<cycle-end id="10" type="scavenge" contextid="4" timestamp="2020-10-18T13:27:09.635" /> 
+
+<allocation-satisfied id="11" threadId="0000000000AC4600" bytesRequested="272" /> 
+
+<af-end id="12" timestamp="2020-10-18T13:27:09.635" threadId="0000000000AC4F80" success="true" from="nursery"/> 
+
+<exclusive-end id="13" timestamp="2020-10-18T13:27:09.635" durationms="31.984" /> 
+```
+You can analyze this *stop-the-world* cycle by inspecting a few tags and xml attributes.  
+
+- **`<exclusive-start>`** The *stop-the-world* cycle stanza begins with a triggering event, represented for *stop-the-world* cycles with a <exclusive-start> tag. The triggering event for the other type of GC cycle, a concurrent cycle, is recorded by the `<concurrent-kickoff>` tag.
+
+- **`<af-start>`** The reason for the triggering of this GC cycle is recorded by the `<af-start>` tag, which indicates that the triggering event is an allocation failure.
+
+- **`<cycle-start>`** The beginning of the GC cycle itself is recorded by the `<cycle-start>` tag and contains the XML attribute `type=scavenge` to describe the GC involved in this GC cycle.  
+
+- **`timestamp`**, **`id`**, and **`context-id`** Each event that is logged in the GC cycle stanza is labeled with a `timestamp`,`id`, and `context-id` XML attribute. The `id` attribute increases incrementally with each event. All gc increments, operations, and concurrent tags associate with a particular cycle have a `contxtid` value that matches the `id` value of the cycle. 
+
+- **`<gc-start>`** The `<gc-start>` tag logs the beginning of a GC increment, in this case of type `scavenge`, nested within the GC cycle. 
```

Comment:
We don't say anything about memory that I can see. Since the cycles are to free up memory, it seems to be important to explain?

i.e. the `<mem>` sections at GC start and GC end? I notice that there is 0% free memory available in the nursery or survivor areas at the start of the cycle, but at the end of the cycle there is free space in the nursery to allocate new objects. Maybe this is covered somewhere else. Ignore if it is, but I couldn't see it.

Reply:
Really good point - I'll work to get this into the worked example to make a fuller example. I am trying to incorporate progreessive disclosure, especially because there are quite a few things to know before you can analyse a cycle, so I may touch on the <mem> info, and take the reader through the full analysis of what is happening to the memory within an increment or cycle in a later example.